### DISCUSSION TITLE
How do I handle circular types?

### DISCUSSION THREAD

**Discussion by Hedwig7s** [teal-language/tl#793] [Q&A]
I have 2 scripts. They both depend on each others' types but only one depends on the other in practice
Because of this I am constantly fighting unnecessary circular dependency errors and it's driving me insane
Is there any way to handle this?

**Comment by hishamhm** [comment]
What I usually do is to break the circular type dependency by creating a third file with the type definitions and requiring it from both using `local type _ = require("_")`.

**Reply by mtdowling** [reply]
I've been running into this issue too recently. This limitation makes it feel like Teal is more geared towards single-file scripts. When you start trying to break monolithic scripts apart to make them easier to read/maintain, you quickly run into this circular dependency issue. I've noticed that [following the docs](https://github.com/teal-language/tl/blob/master/docs/tutorial.md#global-variables) on how to declare these types globally doesn't quite work as I'd expect either (you can refer to a type, but you can't actually use them/call methods from them in the file that uses `global type Foo`).
